tests/src/JIT/SIMD/VectorCast.cs

```diff
@@ -0,0 +1,193 @@
+        succeeded &= Test_SByte  <UInt32> ();
+        succeeded &= Test_UInt16 <UInt32> ();
+        succeeded &= Test_Int16  <UInt32> ();
+        succeeded &= Test_UInt32 <UInt32> ();
+        succeeded &= Test_Int32  <UInt32> ();
+        succeeded &= Test_UInt64 <UInt32> ();
+        succeeded &= Test_Int64  <UInt32> ();
+        succeeded &= Test_Single <UInt32> ();
+        succeeded &= Test_Double <UInt32> ();
+                                 
+        succeeded &= Test_Byte   <Int32>  ();
+        succeeded &= Test_SByte  <Int32>  ();
+        succeeded &= Test_UInt16 <Int32>  ();
+        succeeded &= Test_Int16  <Int32>  ();
+        succeeded &= Test_UInt32 <Int32>  ();
+        succeeded &= Test_Int32  <Int32>  ();
+        succeeded &= Test_UInt64 <Int32>  ();
+        succeeded &= Test_Int64  <Int32>  ();
+        succeeded &= Test_Single <Int32>  ();
+        succeeded &= Test_Double <Int32>  ();
+                                 
+        succeeded &= Test_Byte   <UInt64> ();
+        succeeded &= Test_SByte  <UInt64> ();
+        succeeded &= Test_UInt16 <UInt64> ();
+        succeeded &= Test_Int16  <UInt64> ();
+        succeeded &= Test_UInt32 <UInt64> ();
+        succeeded &= Test_Int32  <UInt64> ();
+        succeeded &= Test_UInt64 <UInt64> ();
+        succeeded &= Test_Int64  <UInt64> ();
+        succeeded &= Test_Single <UInt64> ();
+        succeeded &= Test_Double <UInt64> ();
+                                 
+        succeeded &= Test_Byte   <Int64>  ();
+        succeeded &= Test_SByte  <Int64>  ();
+        succeeded &= Test_UInt16 <Int64>  ();
+        succeeded &= Test_Int16  <Int64>  ();
+        succeeded &= Test_UInt32 <Int64>  ();
+        succeeded &= Test_Int32  <Int64>  ();
+        succeeded &= Test_UInt64 <Int64>  ();
+        succeeded &= Test_Int64  <Int64>  ();
+        succeeded &= Test_Single <Int64>  ();
+        succeeded &= Test_Double <Int64>  ();
+                                 
+        succeeded &= Test_Byte   <Single> ();
+        succeeded &= Test_SByte  <Single> ();
+        succeeded &= Test_UInt16 <Single> ();
+        succeeded &= Test_Int16  <Single> ();
+        succeeded &= Test_UInt32 <Single> ();
+        succeeded &= Test_Int32  <Single> ();
+        succeeded &= Test_UInt64 <Single> ();
+        succeeded &= Test_Int64  <Single> ();
+        succeeded &= Test_Single <Single> ();
+        succeeded &= Test_Double <Single> ();
+                                          
+        succeeded &= Test_Byte   <Double> ();
+        succeeded &= Test_SByte  <Double> ();
+        succeeded &= Test_UInt16 <Double> ();
+        succeeded &= Test_Int16  <Double> ();
+        succeeded &= Test_UInt32 <Double> ();
+        succeeded &= Test_Int32  <Double> ();
+        succeeded &= Test_UInt64 <Double> ();
+        succeeded &= Test_Int64  <Double> ();
+        succeeded &= Test_Single <Double> ();
+        succeeded &= Test_Double <Double> ();
```

Comment:
One problem with this test is that if it fails, you have no idea what failed without running the test in a debugger and (hopefully) seeing the same failure as whoever saw the failure in the first place.

Would it be possible to print out a message when the test fails that indicate which tests fails? This could be as simple as numbering all the test cases, possibly by passing a unique int to the `Test_*` function, and outputting that number if the condition fails, e.g., `Test 23 failed`.